app/rdf/query-dimension-values.ts

```diff
@@ -185,12 +192,17 @@ CONSTRUCT {
   OPTIONAL { ?value schema:color ?color . }
 }`;
 
-  const quads = await sparqlClient.query.construct(query, {
-    operation: "postUrlencoded",
-  });
-  const valuesQuads = quads.filter((q) => q.predicate.equals(ns.rdf.first));
-
-  return valuesQuads.map((qValue) => parseDimensionValue(qValue, quads));
+  return await executeWithCache(
+    sparqlClient,
+    query,
+    () => sparqlClient.query.construct(query, { operation: "postUrlencoded" }),
```

Comment:
Nit: Since execute with cache already receives the sparqlClient, I wonder if it would not be simpler to have a mode:'construct'|'select'|(keyof typeof sparqlClient.query) so that the user here does not have to know about the details ? Isn't it possible to infer from the query that it is a CONSTRUCT or SELECT ?

Reply:
Some of the queries use "raw" SPARQL, without SELECT of CONSTRUCT utilities from Zazuko's libraries. I felt it was easier to write them like this, as otherwise I had some problems with passing `ns` objects inside the CONSTRUCT queries, and also I am not 100% sure if we can control placement of pragmas directly in the middle of a query (as we now do with `#pragma evaluate on`). Let me know if that's a deal-breaker, I can investigate then further in a next PR 🙇‍♂️